From d0a1a4e6767da8a2455b8822446978b35813161d Mon Sep 17 00:00:00 2001
From: Emmanuele Bassi <ebassi@gnome.org>
Date: Tue, 7 Mar 2017 11:52:26 +0000
Subject: [PATCH 3/6] Add explicit version flags for macOS builds

Autotools automatically adds version related linker flags on macOS that
are not related to the version information. Since we want to maintain
binary compatibility, we should do the same when building on macOS.

Fixes: #108
---
 src/meson.build | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/meson.build b/src/meson.build
index ecd032a..4dbb3eb 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -51,10 +51,14 @@ foreach h: headers
   epoxy_headers += join_paths(meson.source_root(), 'include/epoxy/@0@'.format(h))
 endforeach
 
+common_ldflags = []
 if cc.get_id() == 'gcc'
   common_ldflags = [ '-Wl,-Bsymbolic', '-Wl,-z,relro', '-Wl,-z,now', ]
-else
-  common_ldflags = []
+endif
+
+# Maintain compatibility with autotools; see: https://github.com/anholt/libepoxy/issues/108
+if host_system == 'darwin'
+  common_ldflags += [ '-compatibility_version=1', '-current_version=1.0', ]
 endif
 
 epoxy_deps = [ dl_dep, ]
-- 
2.12.0

